vercel/resource_edge_config_item.go

```diff
@@ -0,0 +1,260 @@
+				PlanModifiers: []planmodifier.String{stringplanmodifier.RequiresReplaceIfConfigured(), stringplanmodifier.UseStateForUnknown()},
+			},
+			"key": schema.StringAttribute{
+				Description:   "The ID of the Edge Config store.",
+				Required:      true,
+				PlanModifiers: []planmodifier.String{stringplanmodifier.RequiresReplace()},
+			},
+			"value": schema.StringAttribute{
+				Description: "The value you want to assign to the key.",
+				Required:    true,
+			},
+		},
+	}
+}
+
+type EdgeConfigItem struct {
+	Key          types.String `tfsdk:"key"`
+	Value        types.String `tfsdk:"value"`
+	TeamID       types.String `tfsdk:"team_id"`
+	EdgeConfigID types.String `tfsdk:"edge_config_id"`
+	Token        types.String `tfsdk:"token"`
+}
+
+func responseToEdgeConfigItem(out client.EdgeConfigItem) EdgeConfigItem {
+	return EdgeConfigItem{
+		Key:   types.StringValue(out.Key),
+		Value: types.StringValue(out.Value),
+	}
+}
+
+// Create will create an edgeConfigToken within Vercel.
+// This is called automatically by the provider when a new resource should be created.
+func (r *edgeConfigItemResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan EdgeConfigItem
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	out, err := r.client.CreateEdgeConfigItem(ctx, client.CreateEdgeConfigItemRequest{
+		TeamID:       plan.TeamID.ValueString(),
+		EdgeConfigID: plan.EdgeConfigID.ValueString(),
+		Token:        plan.Token.ValueString(),
+		Key:          plan.Key.ValueString(),
+		Value:        plan.Value.ValueString(),
+	})
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating Edge Config Item",
+			"Could not create Edge Config Item, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	result := responseToEdgeConfigItem(out)
+	tflog.Info(ctx, "created Edge Config Item", map[string]interface{}{
+		"edge_config_id": plan.EdgeConfigID.ValueString(),
+		"key":            result.Key.ValueString(),
+		"value":          result.Value.ValueString(),
+	})
+
+	diags = resp.State.Set(ctx, result)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Read will read edgeConfigToken information by requesting it from the Vercel API, and will update terraform
+// with this information.
+func (r *edgeConfigItemResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state EdgeConfigItem
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	out, err := r.client.GetEdgeConfigItem(ctx, client.EdgeConfigItemRequest{
+		Token:        state.Token.ValueString(),
+		TeamID:       state.TeamID.ValueString(),
+		EdgeConfigID: state.EdgeConfigID.ValueString(),
+		Key:          state.Key.ValueString(),
+		Value:        state.Value.ValueString(),
+	})
```

Comment:
that's a lot of fields for a Get request 🤔

Reply:
Just double checking here, but the minimum set needed here is `EdgeConfigID, TeamID, Key`. Is it okay to not pass `Value` here?